packages/react-native-editor/bin/generate-pot-files.sh

```diff
@@ -0,0 +1,243 @@
+    $WP_CLI --info
+  fi
+}
+
+function generate_bundles() {
+  local ENTRY_FILE=$1
+  local ANDROID_BUNDLE_OUTPUT=$2
+  local ANDROID_SOURCEMAP_OUTPUT=$3
+  local IOS_BUNDLE_OUTPUT=$4
+  local IOS_SOURCEMAP_OUTPUT=$5
+
+  echo -e "\n\033[1mGenerate Android JS bundle\033[0m"
+  $BUNDLE_CLI --platform android --dev false --entry-file "$ENTRY_FILE" --bundle-output "$ANDROID_BUNDLE_OUTPUT" --sourcemap-output "$ANDROID_SOURCEMAP_OUTPUT"
+  
+  echo -e "\n\033[1mGenerate iOS JS bundle\033[0m"
+  $BUNDLE_CLI --platform ios --dev false --entry-file "$ENTRY_FILE" --bundle-output "$IOS_BUNDLE_OUTPUT" --sourcemap-output "$IOS_SOURCEMAP_OUTPUT"
+}
+
+function extract_source_from_sourcemap_file() {
+  local MAP_FILE=$1
+  local TARGET_PATH=$2
+
+  mkdir -p $TARGET_PATH
+
+  echo -e "\n\033[1mExtracting source files from \"$MAP_FILE\" source map file\033[0m"
+  node $SCRIPT_DIR/extract-files-from-sourcemap.js $MAP_FILE $TARGET_PATH
+}
+
+function generate_pot_files() {
+  local OUTPUT_PATH=$1
+  local PLUGIN_NAME=$2
+  local SOURCE_DIR=$3
+  shift 3
+  local PLUGINS_TO_SUBTRACT=( $@ )
+
+  # Define subtract pot files
+  local SUBTRACT_POT_FILES=()
+  for PLUGIN in "$@"; do
+    SUBTRACT_POT_FILES+=( "$OUTPUT_PATH/$PLUGIN-used-android.pot" )
+    SUBTRACT_POT_FILES+=( "$OUTPUT_PATH/$PLUGIN-used-ios.pot" )
+  done
+  local SUBTRACT_POT_FILES=$(join_by , ${SUBTRACT_POT_FILES[@]})
+
+  # Define output paths
+  local OUTPUT_POT_USED_ANDROID_FILE="$OUTPUT_PATH/$PLUGIN_NAME-used-android.pot"
+  local OUTPUT_POT_USED_IOS_FILE="$OUTPUT_PATH/$PLUGIN_NAME-used-ios.pot"
+  local OUTPUT_POT_BLOCKS_FILE="$OUTPUT_PATH/$PLUGIN_NAME-blocks.pot"
+  local OUTPUT_POT_SOURCE_FILE="$OUTPUT_PATH/$PLUGIN_NAME-source.pot"
+
+  local EXCLUDE_FILES="test/*,e2e-tests/*,bundle/*,build-module/*"
+
+  local DEBUG_PARAM=$([ -z $DEBUG ] && echo "" || echo "--debug")
+  local SUBTRACT_PARAM=$([ -z $SUBTRACT_POT_FILES ] && echo "" || echo "--subtract=$SUBTRACT_POT_FILES")
+  local DOMAIN_PARAM=$([ "$PLUGIN_NAME" == "gutenberg" ] && echo "--ignore-domain" || echo "--domain=$PLUGIN_NAME")
+
+  local MAKEPOT_COMMAND="$WP_CLI i18n make-pot"
+
+  echo -e "\n\033[1mExtract strings and generate POT files for \"$PLUGIN_NAME\" plugin from \"$SOURCE_DIR\"\033[0m"
+
+  mkdir -p $OUTPUT_PATH
+
+  if [ -n "$SUBTRACT_POT_FILES" ]; then
+    echo "--- Strings from ${PLUGINS_TO_SUBTRACT[@]} plugins will be subtracted ---"
+  fi
+  
+  echo -e "\nExtract used strings from Android source-map:"
+  $MAKEPOT_COMMAND $ANDROID_EXTRACT_SOURCE_FILES_PATH $DEBUG_PARAM $SUBTRACT_PARAM $DOMAIN_PARAM $OUTPUT_POT_USED_ANDROID_FILE
+
+  echo -e "\nExtract used strings from iOS source-map:"
+  $MAKEPOT_COMMAND $IOS_EXTRACT_SOURCE_FILES_PATH $DEBUG_PARAM $SUBTRACT_PARAM $DOMAIN_PARAM $OUTPUT_POT_USED_IOS_FILE
+
+  echo -e "\nExtract strings from block JSON files:"
+  $MAKEPOT_COMMAND $SOURCE_DIR $DEBUG_PARAM --exclude="$EXCLUDE_FILES" --skip-js --skip-php --ignore-domain $OUTPUT_POT_BLOCKS_FILE
```

Comment:
Usually, the block JSON files are being parsed if they're contained in the source directory passed, however, I noticed that this is only the case when ignoring the domain. For this reason, I decided to extract it as an extra step and merge it when extracting the strings from non-native JS code.